netcam_rtsp.c

```diff
@@ -456,6 +452,23 @@ static int netcam_rtsp_open_context(netcam_context_ptr netcam){
         return -1;
     }
 
+    /*
+     * Documentation does not indicate thread safety on these
+     * init functions so we lock them just in case.
+     */
+    if (netcam->rtsp->status == RTSP_RECONNECTING){
+        pthread_mutex_lock(&global_lock);
+            avformat_network_deinit();
```

Comment:
Hm, this doesn't feel right to me. We could have other netcam threads using avformat's network support, removing it whilst they're running sounds potentially bad.

I understand how you got here though. I'm very puzzled as to why we get told avformat_network_init hasn't been called. I think there may be some more complicated explanation.